prisma-fmt/src/merge_schemas.rs

```diff
@@ -0,0 +1,127 @@
+use psl::reformat_validated_schema_into_single;
+use serde::Deserialize;
+
+use crate::schema_file_input::SchemaFileInput;
+
+#[derive(Debug, Deserialize)]
+pub struct MergeSchemasParams {
+    schema: SchemaFileInput,
+}
+
+pub(crate) fn merge_schemas(params: &str) -> String {
+    let params: MergeSchemasParams = match serde_json::from_str(params) {
+        Ok(params) => params,
+        Err(serde_err) => {
+            panic!("Failed to deserialize GetDmmfParams: {serde_err}");
+        }
+    };
+
+    let params_as_vec: Vec<_> = params.schema.into();
+    let validated_schema = psl::validate_multi_file(params_as_vec.clone());
+
+    // diagnostics aren't supposed to have errors, as they should be validated before-hand.
+    if validated_schema.diagnostics.has_errors() {
+        panic!("Invalid schemas.");
```

Comment:
I think we should return proper rendered validation message here (similar to the one `validate` returns), not just panic

Reply:
`merge_schemas` will only be called by `@prisma/client`, correct?
`prisma validate` is implicitly run by the Client's generator build process.

In a scenario with multiple schemas, we'd let the `@prisma/client` build generator run:
- `prisma validate` on the multiple schemas the user defined
- If the schemas are valid, it'd merge them into a single one that is guaranteed to be valid.

The snippet highlighted above has a panic as it indicates the violation of a precondition.
Do you reckon we should discard the precondition entirely, and behave like `validate` in here?

Reply:
(Context on [Notion](https://www.notion.so/prismaio/Multi-file-Schema-24d68fe8664048ad86252fe446caac24?pvs=4#2d3b739b949c4043a9c27e95d6ad9f14))

Reply:
I think it would be safer to not assume anything about input from our end. We can also think about completely replacing `validate` with `mergeSchemas` in the client once feature goes GA.

Reply:
Cool. Done in https://github.com/prisma/prisma-engines/pull/4787/commits/c34d7b5f5430ffe0e7755cf441d6e8e75c46c06a.
I've also updated https://github.com/prisma/prisma-engines/pull/4787#issuecomment-2039349964.